crucible-mir/src/Mir/Trans.hs

```diff
@@ -1266,6 +1396,10 @@ evalPlaceProj ty pl@(MirPlace tpr ref NoMeta) M.Deref = do
     doRef (M.TySlice ty') | MirSliceRepr <- tpr = doSlice ty' ref
     doRef M.TyStr | MirSliceRepr <- tpr = doSlice (M.TyUint M.B8) ref
     doRef (M.TyDynamic _) | DynRefRepr <- tpr = doDyn ref
+    doRef adtTy@(M.TyAdt _ _ _) | MirSliceRepr <- tpr = doSliceAdt adtTy ref
+    doRef adtTy@(M.TyAdt _ _ _) | DynRefRepr <- tpr = doDynAdt adtTy ref
+    -- TODO: do we need to go to the trouble here of figuring out if this ADT is
+    -- a DST, if it didn't come with a DST-indicative TypeRepr?
```

Comment:
Opinions welcome. It wouldn't be too hard to check, but I think that such a thing ought to be an error, and all that checking would get us would be a very slightly more useful error message than `doRef`'s fallback case.